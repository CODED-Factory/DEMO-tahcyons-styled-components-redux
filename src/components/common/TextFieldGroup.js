import React from "react";
import PropTypes from "prop-types";

const TextFieldGroup = ({
  name,
  value,
  error,
  placeholder,
  info,
  type,
  label,
  onChange,
  required,
  checked,
  disabled
}) => {
  const requiredSpan = (
    <span className={`normal ${required ? "red" : "black-60"}`}>
      {required ? " *" : " (optional)"}
    </span>
  );
  return (
    <div className="measure mt3">
      {label && (
        <label htmlFor={name} className="db fw6 lh-copy f6">
          {label}
          {requiredSpan}
        </label>
      )}
      <input
        type={type}
        className={`pa2 input-reset ba hover-bg-black hover-white w-100 ${
          error ? "b--red" : " b--black-20"
        }`}
        placeholder={placeholder}
        name={name}
        value={value}
        onChange={onChange}
        disabled={disabled}
        checked={checked}
        id={name}
        aria-describedby={info ? "name-desc" : name}
      />
      {info && (
        <small id="name-desc" className="f6 black-60 db mb2">
          {info}
        </small>
      )}
      {error && <small className="f6 red db mb2">{error}</small>}
    </div>
  );
};

TextFieldGroup.propTypes = {
  name: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  label: PropTypes.string,
  placeholder: PropTypes.string,
  info: PropTypes.string,
  error: PropTypes.oneOfType([PropTypes.string, PropTypes.array]),
  disabled: PropTypes.bool,
  required: PropTypes.bool
};

TextFieldGroup.defaultProps = {
  type: "text"
};

export default TextFieldGroup;
