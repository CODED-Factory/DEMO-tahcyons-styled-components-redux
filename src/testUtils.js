import casual from "casual";

casual.seed(666);

export const fakeUserData = overrides => ({
  username: casual.username,
  password: casual.password,
  ...overrides
});

export const type = (wrapper, name, value) => {
  wrapper.find(`input[name="${name}"]`).simulate("change", {
    target: { name, value }
  });
};
