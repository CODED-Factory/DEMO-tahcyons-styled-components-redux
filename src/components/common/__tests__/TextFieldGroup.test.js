import React from "react";
import { shallow } from "enzyme";

// Components
import TextFieldGroup from "../TextFieldGroup";

// Test Utils
import { type } from "../../../testUtils";

describe("TextFieldGroup", () => {
  describe("rendering", () => {
    it("renders with basic props", () => {
      const wrapper = shallow(
        <TextFieldGroup
          name="password"
          type="password"
          value="password123"
          label="Password"
          onChange={() => {}}
        />
      );
      expect(wrapper).toMatchSnapshot();
    });
    it("renders with optional props", () => {
      const wrapper = shallow(
        <TextFieldGroup
          name="password"
          type="password"
          value=""
          label="Password"
          required
          disabled
          info="put yo password"
          placeholder="password123"
          error="y u no password?!"
          onChange={() => {}}
        />
      );
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe("interaction", () => {
    it("runs the onChange function", () => {
      const onChange = jest.fn();
      const wrapper = shallow(
        <TextFieldGroup
          name="password"
          type="password"
          value="password123"
          label="Password"
          onChange={onChange}
        />
      );
      type(wrapper, "password", "password123");
      expect(onChange).toHaveBeenCalled();
    });
  });
});
