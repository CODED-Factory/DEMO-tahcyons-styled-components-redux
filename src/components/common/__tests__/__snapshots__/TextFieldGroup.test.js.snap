// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextFieldGroup rendering renders with basic props 1`] = `
<div
  className="measure mt3"
>
  <label
    className="db fw6 lh-copy f6"
    htmlFor="password"
  >
    Password
    <span
      className="normal black-60"
    >
       (optional)
    </span>
  </label>
  <input
    aria-describedby="password"
    className="pa2 input-reset ba hover-bg-black hover-white w-100  b--black-20"
    id="password"
    name="password"
    onChange={[Function]}
    type="password"
    value="password123"
  />
</div>
`;

exports[`TextFieldGroup rendering renders with optional props 1`] = `
<div
  className="measure mt3"
>
  <label
    className="db fw6 lh-copy f6"
    htmlFor="password"
  >
    Password
    <span
      className="normal red"
    >
       *
    </span>
  </label>
  <input
    aria-describedby="name-desc"
    className="pa2 input-reset ba hover-bg-black hover-white w-100 b--red"
    disabled={true}
    id="password"
    name="password"
    onChange={[Function]}
    placeholder="password123"
    type="password"
    value=""
  />
  <small
    className="f6 black-60 db mb2"
    id="name-desc"
  >
    put yo password
  </small>
  <small
    className="f6 red db mb2"
  >
    y u no password?!
  </small>
</div>
`;
